### ISSUE TITLE
Tooltip remains visible on elements that receive focus until user clicks/moves focus away

### ISSUE THREAD

**Issue by alvarotrigo** [twbs/bootstrap#18777]
Bootstrap tooltip doesn't seem to be working as expected when there's a checkbox.

```html
<label class="checkbox" data-toggle="tooltip" data-placement="bottom" title="Demo Text">
  <input type="checkbox" value="1" name="rememberMe" /> Remember me
</label>
```
```javascript
$(function() {
  $('[data-toggle="tooltip"]').tooltip()
})

```
[Reproduction online](https://jsfiddle.net/0nokdhr6/2/)

[This answer on stackoverflow](http://stackoverflow.com/a/32915599/1081396) seems to be a working solution.

I'm not aware of the behavior of this in Bootstrap 4.

**Comment by patrickhlauke** [comment]
that jsfiddle works for me in chrome, firefox, ie11 and edge in windows. in what situations is it not working for you?

**Comment by alvarotrigo** [comment]
I should have explained it better.
It works good when you hover the element. Now, when you click the checkbox, then the tooltip will always stay visible no matter you hove out.

**Comment by alvarotrigo** [comment]
The issue is also mentioned in other stackoverflow questions such as:
http://stackoverflow.com/q/32915155/1081396
http://stackoverflow.com/q/32095144/1081396

**Comment by patrickhlauke** [comment]
Gotcha. This is normal (though annyoing, admittedly) behavior due to the fact that tooltips must also be triggered for keyboard users, so showing/hiding them is tied to element focus as well as mouse hover. When you click on a checkbox (or a `<button>`, for instance), you're also moving focus to it...and while it has focus, the tooltip will remain visible.

**Comment by mdo** [comment]
[**Bootstrap 3 is no longer being officially developed or supported.**](https://github.com/twbs/bootstrap/issues/20631)

All work has moved onto our next major release, v4. As such, this issue or pull request is being closed as a "won't fix." For additional help and support, we recommend [utilizing our community resources](https://github.com/twbs/bootstrap/blob/master/README.md#community). Thanks for your understanding, and see you on the other side of v4!

<3,
@mdo and team
